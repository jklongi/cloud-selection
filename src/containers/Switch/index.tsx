import FormControl from '@material-ui/core/FormControl'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import FormHelperText from '@material-ui/core/FormHelperText'
import MuiSwitch, { SwitchProps } from '@material-ui/core/Switch'
import React from 'react'

type Props = Omit<SwitchProps, 'onChange'> & {
    label: string
    onChange: (selected: boolean) => void
    helperText?: string
}

const Switch = ({ label, helperText, onChange, ...rest }: Props) => {
    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        onChange(event.target.checked)
    }
    return (
        <FormControl>
            <FormControlLabel control={<MuiSwitch onChange={handleChange} color="primary" {...rest} />} label={label} />
            <FormHelperText>{helperText}</FormHelperText>
        </FormControl>
    )
}

export default Switch
