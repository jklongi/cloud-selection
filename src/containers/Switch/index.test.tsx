import { mount } from 'enzyme'
import Switch from './index'

describe('<Switch />', () => {
    it('matches snapshot', () => {
        const wrapper = mount(<Switch onChange={() => {}} label="Label" helperText="Helper" />)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders label', () => {
        const wrapper = mount(<Switch onChange={() => {}} label="Label" />)
        expect(wrapper.find('label').text()).toEqual('Label')
    })
    it('renders helper text', () => {
        const wrapper = mount(<Switch onChange={() => {}} label="Label" helperText="Helper" />)
        expect(wrapper.find('p').text()).toEqual('Helper')
    })
    it('input changes trigger callback', () => {
        const onChange = jest.fn()
        const wrapper = mount(<Switch checked={false} onChange={onChange} label="Label" />)
        const input = wrapper.find('input')
        input.simulate('change', { target: { checked: true } })
        expect(onChange).toHaveBeenCalled()
    })

    it('callback value is boolean', () => {
        let state = false
        const onChange = (checked: boolean) => {
            state = checked
        }
        const wrapper = mount(<Switch checked={state} onChange={onChange} label="Label" />)
        const input = wrapper.find('input')
        input.simulate('change', { target: { checked: true } })
        expect(state).toEqual(true)
    })
})
