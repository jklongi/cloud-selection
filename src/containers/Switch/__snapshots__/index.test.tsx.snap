// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> matches snapshot 1`] = `ReactWrapper {}`;
